src/charmhub_listing_review/update_issue.py

```diff
@@ -0,0 +1,458 @@
+""".strip()  # noqa: E501
+    )
+
+    return ''.join(description)
+
+
+# TODO: It would be better to amend the docs so that we don't have these issues
+# than to have a manually curated set of practices to ignore.
+IGNORED_BEST_PRACTICES = {
+    # This is covered by the more extensive items above. It's also duplicated in
+    # the docs.
+    '* [ ] The quality assurance pipeline of a charm should be automated using a '
+    'continuous integration (CI) system.',
+    # This is duplicated by another best practice note.
+    "* [ ] If you're setting up a ``git`` repository: name it using the pattern "
+    '``<charm name>-operator``. For the charm name, see :ref:`specify-a-name`.',
+    # These don't seem like best practice notes -- we should adjust the docs.
+    '* [ ] Smaller charm documentation examples:',
+    '* [ ] Bigger charm documentation examples:',
+}
+
+
+def extract_best_practice_blocks(file_path: pathlib.Path):
+    """Extracts 'Best practice' blocks from a file."""
+    remove_pattern: str | None = None
+    matches: list[str] = []
+    content = file_path.read_text()
+    if file_path.suffix == '.md':
+        matches = BEST_PRACTICE_RE_MD.findall(content)
+        remove_pattern = r'^:class: hint\s*\n'
+    elif file_path.suffix == '.rst':
+        matches = BEST_PRACTICE_RE_REST.findall(content)
+        remove_pattern = r'^\s+'
+    assert remove_pattern is not None, 'Unsupported file type for best practices extraction.'
+    if not matches:
+        return matches
+    return [
+        re.sub(remove_pattern, '', match, flags=re.MULTILINE).strip().replace('\n', ' ')
+        for match in matches
+    ]
+
+
+def find_best_practices(path_to_ops: pathlib.Path, path_to_charmcraft: pathlib.Path):
+    """Recursively located best practice blocks in Ops and Charmcraft."""
+    checklist: list[str] = []
+    for directory in (path_to_ops, path_to_charmcraft):
+        for file_path in directory.rglob('*'):
+            if file_path.suffix in ('.md', '.rst'):
+                practices = (
+                    practice
+                    for practice in extract_best_practice_blocks(file_path)
+                    if practice not in IGNORED_BEST_PRACTICES
+                )
+                checklist.extend(practices)
+    checklist.sort()
+    return checklist
+
+
+class _IssueData(TypedDict):
+    """Typed dictionary for issue data."""
+
+    name: str
+    demo_url: str
+    project_repo: str
+    ci_linting: str
+    ci_release_url: str
+    ci_integration_url: str
+    documentation_link: str
+    contribution_link: str
+    license_link: str
+    security_link: str
+
+
+def get_details_from_issue(issue_number: int):
+    """Fetch details from the issue number using the GitHub CLI.
+
+    Requires `gh` CLI to be installed and authenticated, and `jq` to be
```

Comment:
Not sure if it's worth it, but we could easily avoid the jq requirement by using json.loads and picking out the "body" field.